src/transformers/models/trillsson_efficient/modeling_trillsson_efficient.py

```diff
@@ -0,0 +1,686 @@
+            in_channels=in_channels, out_channels=out_channels, kernel_size=1, stride=1, padding=0, bias=False
+        )
+        self.top_bn = nn.BatchNorm2d(num_features=out_channels, eps=norm_eps, momentum=norm_momentum)
+        self.activation = ACT2FN[config.hidden_act]
+
+    def forward(self, x):
+        x = self.top_conv(x)
+        x = self.top_bn(x)
+        x = self.activation(x)
+        return x
+
+
+class MBConvBlock(nn.Module):
+    def __init__(
+        self,
+        config: Trillsson_efficientConfig,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        expand_ratio: int,
+        use_se: int,
+        survival_probability: int,
+        norm_eps: float,
+        norm_momentum: float,
+    ) -> None:
+        """MBConv block: Mobile Inverted Residual Bottleneck."""
+        super().__init__()
+        hidden_dim = round(in_channels * expand_ratio)
+        self.config = config
+        self.expand_ratio = expand_ratio
+        self.use_se = use_se
+        self.survival_probability = survival_probability
+        self.identity = stride == 1 and in_channels == out_channels
+        self.activation = ACT2FN[config.hidden_act]
+        self.dropout = nn.Dropout(self.survival_probability)
+        if self.expand_ratio != 1:
+            self.expand_conv = Conv2DSamePadding(
+                in_channels=in_channels, out_channels=hidden_dim, kernel_size=1, stride=1, padding=0, bias=False
+            )
+            self.expand_bn = nn.BatchNorm2d(num_features=hidden_dim, eps=norm_eps, momentum=norm_momentum)
+        self.dwconv2 = Conv2DSamePadding(
+            in_channels=hidden_dim,
+            out_channels=hidden_dim,
+            kernel_size=3,
+            stride=stride,
+            padding=0,
+            groups=hidden_dim,
+            bias=False,
+        )
+        self.bn = nn.BatchNorm2d(num_features=hidden_dim, eps=norm_eps, momentum=norm_momentum)
+        if self.use_se:
+            self.se = SELayer(config=self.config, in_channels=in_channels, out_channels=hidden_dim)
+            self.project_conv = Conv2DSamePadding(
+                in_channels=hidden_dim, out_channels=out_channels, kernel_size=1, stride=1, padding=0, bias=False
+            )
+            self.project_bn = nn.BatchNorm2d(num_features=out_channels, eps=norm_eps, momentum=norm_momentum)
+
+    def forward(self, input):
+        if self.expand_ratio != 1:
+            x = self.expand_conv(input)
+            x = self.expand_bn(x)
+            x = self.activation(x)
+        else:
+            x = input
+        x = self.dwconv2(x)
+        x = self.bn(x)
+        x = self.activation(x)
+        if self.use_se:
+            x = self.se(x)
+            x = self.project_conv(x)
+            x = self.project_bn(x)
+            if self.identity:
+                if self.survival_probability:
+                    x = self.dropout(x)
+                x = input + x
+        return x
```

Comment:
No single letter variable names, lets try to make it more readble and understandable ! 😉